rpm/0005-Use-mce-for-keyboard-state.-Fixes-JB-48910.patch

```diff
@@ -88,44 +137,26 @@ index b7126b5..9cb326d 100644
 +#ifdef HAVE_UDEV
  void MImHwKeyboardTrackerPrivate::detectEvdev()
  {
--#ifndef HAVE_CONTEXTSUBSCRIBER
      // Use udev to enumerate all input devices, using evdev on each device to
-     // find the first device offering a SW_TABLET_MODE switch. If found, this
-     // switch is used to determine keyboard presence.
-@@ -107,12 +129,10 @@ void MImHwKeyboardTrackerPrivate::detectEvdev()
-     }
-     udev_enumerate_unref(enumerate);
-     udev_unref(udev);
--#endif
- }
- 
- void MImHwKeyboardTrackerPrivate::evdevEvent()
- {
--#ifndef HAVE_CONTEXTSUBSCRIBER
-     // Parse the evdev event and look for SW_TABLET_MODE status.
- 
-     struct input_event ev;
-@@ -130,14 +150,12 @@ void MImHwKeyboardTrackerPrivate::evdevEvent()
+@@ -128,11 +151,12 @@ void MImHwKeyboardTrackerPrivate::evdevEvent()
          evdevTabletModePending = -1;
          Q_EMIT stateChanged();
      }
--#endif
+-
  }
  
  void MImHwKeyboardTrackerPrivate::tryEvdevDevice(const char *device)
  {
-     Q_UNUSED(device);
- 
--#ifndef HAVE_CONTEXTSUBSCRIBER
++    Q_UNUSED(device);
```

Comment:
Is this change needed? It is within an ifdef which I assume disables this whole function in our case.

Reply:
There was three commits touching hw keyboard tracking. Couple two doing first small adjustments to context subscriber path and the last one for switching to mce. Now I just merged these together and didn't bother removing unnecessary content as there didn't appear to be too much of it.

Indeed this could be skipped. Maybe I'll do that if I end up updating the PR for some other reason.

Reply:
Done.